spring-cloud-netflix-zuul/src/main/java/org/springframework/cloud/netflix/zuul/filters/route/SimpleHostRoutingFilter.java

```diff
@@ -137,6 +138,9 @@ public void onPropertyChange(EnvironmentChangeEvent event) {
 				catch (IOException ex) {
 					log.error("error closing client", ex);
 				}
+				// Re-create connection manager (may be shut down on HTTP client close)
+				SimpleHostRoutingFilter.this.connectionManager.shutdown();
```

Comment:
This should move in the above try catch or in its own.

Reply:
`connectionManager.shutdown()` doesn't throw `IOException`, but `httpClient.close()` does.

Is there need to catch `RuntimeException` for `connectionManager.shutdown()` ?

Reply:
you're correct.

Reply:
Wrapped it in e4f3f60addd4e77e567fecc6dd77c96e6a5e0994